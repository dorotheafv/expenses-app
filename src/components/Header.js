import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
    <header>
        <h1>Expensify</h1>
        <div>
            <NavLink to="/" exact={true} activeClassName="is-active">ExpensesDashboardPage</NavLink>
        </div>
        <div>
            <NavLink to="/create" activeClassName="is-active">Create</NavLink>
        </div>
        <div>
            <NavLink to="/edit" activeClassName="is-active">Edit</NavLink>
        </div>
        <div>
            <NavLink to="/help" activeClassName="is-active">Help</NavLink>
        </div>

    </header>
);

export default Header;