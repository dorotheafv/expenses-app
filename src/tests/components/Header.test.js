import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../components/Header';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import toJSON from 'enzyme-to-json';

test('should render Header correctly with shallow',()=>{
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h1').length).toBe(1);
    expect(wrapper.find('h1').text()).toBe('Expensify');
  
    });