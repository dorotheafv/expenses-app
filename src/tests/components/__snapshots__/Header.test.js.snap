// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly with shallow 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <div>
    <NavLink
      activeClassName="is-active"
      exact={true}
      to="/"
    >
      ExpensesDashboardPage
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName="is-active"
      to="/create"
    >
      Add
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName="is-active"
      to="/edit"
    >
      Edit
    </NavLink>
  </div>
  <div>
    <NavLink
      activeClassName="is-active"
      to="/help"
    >
      Help
    </NavLink>
  </div>
</header>
`;
